### YamlMime:ManagedReference
items:
- uid: System.Workflow.Activities.Rules.RuleEvaluationException
  id: RuleEvaluationException
  children:
  - System.Workflow.Activities.Rules.RuleEvaluationException.#ctor
  - System.Workflow.Activities.Rules.RuleEvaluationException.#ctor(System.String)
  - System.Workflow.Activities.Rules.RuleEvaluationException.#ctor(System.Runtime.Serialization.SerializationInfo,System.Runtime.Serialization.StreamingContext)
  - System.Workflow.Activities.Rules.RuleEvaluationException.#ctor(System.String,System.Exception)
  langs:
  - csharp
  name: RuleEvaluationException
  nameWithType: RuleEvaluationException
  fullName: System.Workflow.Activities.Rules.RuleEvaluationException
  type: Class
  summary: "表示由规则计算问题所导致的所有异常的基类。"
  syntax:
    content: 'public class RuleEvaluationException : System.Workflow.Activities.Rules.RuleException'
  inheritance:
  - System.Exception
  - System.Workflow.Activities.Rules.RuleException
  implements: []
  inheritedMembers: []
  platform:
  - net462
- uid: System.Workflow.Activities.Rules.RuleEvaluationException.#ctor
  id: '#ctor'
  parent: System.Workflow.Activities.Rules.RuleEvaluationException
  langs:
  - csharp
  name: RuleEvaluationException()
  nameWithType: RuleEvaluationException.RuleEvaluationException()
  fullName: System.Workflow.Activities.Rules.RuleEvaluationException.RuleEvaluationException()
  type: Constructor
  assemblies:
  - System.Workflow.Activities
  namespace: System.Workflow.Activities.Rules
  summary: "初始化的新实例<xref href=&quot;System.Workflow.Activities.Rules.RuleEvaluationException&quot;></xref>类。"
  remarks: "此构造函数初始化<xref:System.Exception.Message%2A>属性为系统提供的消息的描述错误并将当前系统区域性考虑在内的新实例。</xref:System.Exception.Message%2A>       下表显示<xref:System.Workflow.Activities.Rules.RuleEvaluationException>.</xref:System.Workflow.Activities.Rules.RuleEvaluationException>实例的初始属性的值      |属性 |值 |  |--------------|-----------|  |<xref:System.Exception.InnerException%2A>|空引用 (`Nothing`在 Visual Basic 中)。 |  |<xref:System.Exception.Message%2A>|系统提供的一个本地化的描述。 |</xref:System.Exception.Message%2A> </xref:System.Exception.InnerException%2A>"
  syntax:
    content: public RuleEvaluationException ();
    parameters: []
  overload: System.Workflow.Activities.Rules.RuleEvaluationException.#ctor*
  exceptions: []
  platform:
  - net462
- uid: System.Workflow.Activities.Rules.RuleEvaluationException.#ctor(System.String)
  id: '#ctor(System.String)'
  parent: System.Workflow.Activities.Rules.RuleEvaluationException
  langs:
  - csharp
  name: RuleEvaluationException(String)
  nameWithType: RuleEvaluationException.RuleEvaluationException(String)
  fullName: System.Workflow.Activities.Rules.RuleEvaluationException.RuleEvaluationException(String)
  type: Constructor
  assemblies:
  - System.Workflow.Activities
  namespace: System.Workflow.Activities.Rules
  summary: "初始化的新实例<xref href=&quot;System.Workflow.Activities.Rules.RuleEvaluationException&quot;></xref>与指定的错误消息的类。"
  remarks: "此构造函数初始化<xref:System.Exception.Message%2A>属性的新实例使用`message`参数。</xref:System.Exception.Message%2A>       下表显示<xref:System.Workflow.Activities.Rules.RuleEvaluationException>.</xref:System.Workflow.Activities.Rules.RuleEvaluationException>实例的初始属性的值      |属性 |值 |  |--------------|-----------|  |<xref:System.Exception.InnerException%2A>|空引用 (`Nothing`在 Visual Basic 中)。 |  |<xref:System.Exception.Message%2A>|错误消息字符串。 |</xref:System.Exception.Message%2A> </xref:System.Exception.InnerException%2A>"
  syntax:
    content: public RuleEvaluationException (string message);
    parameters:
    - id: message
      type: System.String
      description: "描述错误的消息。"
  overload: System.Workflow.Activities.Rules.RuleEvaluationException.#ctor*
  exceptions: []
  platform:
  - net462
- uid: System.Workflow.Activities.Rules.RuleEvaluationException.#ctor(System.Runtime.Serialization.SerializationInfo,System.Runtime.Serialization.StreamingContext)
  id: '#ctor(System.Runtime.Serialization.SerializationInfo,System.Runtime.Serialization.StreamingContext)'
  parent: System.Workflow.Activities.Rules.RuleEvaluationException
  langs:
  - csharp
  name: RuleEvaluationException(SerializationInfo,StreamingContext)
  nameWithType: RuleEvaluationException.RuleEvaluationException(SerializationInfo,StreamingContext)
  fullName: System.Workflow.Activities.Rules.RuleEvaluationException.RuleEvaluationException(SerializationInfo,StreamingContext)
  type: Constructor
  assemblies:
  - System.Workflow.Activities
  namespace: System.Workflow.Activities.Rules
  summary: "初始化的新实例<xref href=&quot;System.Workflow.Activities.Rules.RuleEvaluationException&quot;></xref>用序列化数据的类。"
  remarks: "在反序列化，以便重建通过流传输的异常对象过程中调用此构造函数。"
  syntax:
    content: protected RuleEvaluationException (System.Runtime.Serialization.SerializationInfo serializeInfo, System.Runtime.Serialization.StreamingContext context);
    parameters:
    - id: serializeInfo
      type: System.Runtime.Serialization.SerializationInfo
      description: "<xref href=&quot;System.Runtime.Serialization.SerializationInfo&quot;> </xref>它保存关于所引发异常的序列化的对象数据。"
    - id: context
      type: System.Runtime.Serialization.StreamingContext
      description: "<xref:System.Runtime.Serialization.StreamingContext>它包含有关源或目标的上下文信息。</xref:System.Runtime.Serialization.StreamingContext>"
  overload: System.Workflow.Activities.Rules.RuleEvaluationException.#ctor*
  exceptions: []
  platform:
  - net462
- uid: System.Workflow.Activities.Rules.RuleEvaluationException.#ctor(System.String,System.Exception)
  id: '#ctor(System.String,System.Exception)'
  parent: System.Workflow.Activities.Rules.RuleEvaluationException
  langs:
  - csharp
  name: RuleEvaluationException(String,Exception)
  nameWithType: RuleEvaluationException.RuleEvaluationException(String,Exception)
  fullName: System.Workflow.Activities.Rules.RuleEvaluationException.RuleEvaluationException(String,Exception)
  type: Constructor
  assemblies:
  - System.Workflow.Activities
  namespace: System.Workflow.Activities.Rules
  summary: "初始化的新实例<xref href=&quot;System.Workflow.Activities.Rules.RuleEvaluationException&quot;></xref>类，使用指定的错误消息和对内部<xref:System.Exception>这是此<xref:System.Exception>。</xref:System.Exception>根源</xref:System.Exception>的引用"
  remarks: "<xref:System.Exception>引发作为以前的直接结果<xref:System.Exception>应包含对前一个异常中的引用<xref:System.Exception.InnerException%2A>属性。</xref:System.Exception.InnerException%2A> </xref:System.Exception> </xref:System.Exception> <xref:System.Exception.InnerException%2A>属性返回相同的值传递到构造函数中，则为空引用 (`Nothing`) 如果<xref:System.Exception.InnerException%2A>属性不提供内部<xref:System.Exception>给构造函数的值。</xref:System.Exception> </xref:System.Exception.InnerException%2A> </xref:System.Exception.InnerException%2A>       下表显示<xref:System.Workflow.Activities.Rules.RuleEvaluationException>.</xref:System.Workflow.Activities.Rules.RuleEvaluationException>实例的初始属性的值      |属性 |值 |  |--------------|-----------|  |<xref:System.Exception.InnerException%2A>|内部<xref:System.Exception>引用。 |  |<xref:System.Exception.Message%2A>|错误消息字符串。 |</xref:System.Exception.Message%2A> </xref:System.Exception> </xref:System.Exception.InnerException%2A>"
  syntax:
    content: public RuleEvaluationException (string message, Exception ex);
    parameters:
    - id: message
      type: System.String
      description: "描述错误的消息。"
    - id: ex
      type: System.Exception
      description: "<xref:System.Exception>导致当前<xref:System.Exception>.</xref:System.Exception> </xref:System.Exception> 如果`innerException`参数不是 null 引用 (<xref uid=&quot;langword_csharp_Nothing&quot; name=&quot;Nothing&quot; href=&quot;&quot;> </xref>在 Visual Basic 中)，<xref:System.Exception>用于处理内部<xref:System.Exception>.</xref:System.Exception>的 catch 块中引发</xref:System.Exception>当前"
  overload: System.Workflow.Activities.Rules.RuleEvaluationException.#ctor*
  exceptions: []
  platform:
  - net462
references:
- uid: System.Workflow.Activities.Rules.RuleException
  isExternal: false
  name: System.Workflow.Activities.Rules.RuleException
- uid: System.Workflow.Activities.Rules.RuleEvaluationException.#ctor
  parent: System.Workflow.Activities.Rules.RuleEvaluationException
  isExternal: false
  name: RuleEvaluationException()
  nameWithType: RuleEvaluationException.RuleEvaluationException()
  fullName: System.Workflow.Activities.Rules.RuleEvaluationException.RuleEvaluationException()
- uid: System.Workflow.Activities.Rules.RuleEvaluationException.#ctor(System.String)
  parent: System.Workflow.Activities.Rules.RuleEvaluationException
  isExternal: false
  name: RuleEvaluationException(String)
  nameWithType: RuleEvaluationException.RuleEvaluationException(String)
  fullName: System.Workflow.Activities.Rules.RuleEvaluationException.RuleEvaluationException(String)
- uid: System.String
  parent: System
  isExternal: true
  name: String
  nameWithType: String
  fullName: System.String
- uid: System.Workflow.Activities.Rules.RuleEvaluationException.#ctor(System.Runtime.Serialization.SerializationInfo,System.Runtime.Serialization.StreamingContext)
  parent: System.Workflow.Activities.Rules.RuleEvaluationException
  isExternal: false
  name: RuleEvaluationException(SerializationInfo,StreamingContext)
  nameWithType: RuleEvaluationException.RuleEvaluationException(SerializationInfo,StreamingContext)
  fullName: System.Workflow.Activities.Rules.RuleEvaluationException.RuleEvaluationException(SerializationInfo,StreamingContext)
- uid: System.Runtime.Serialization.SerializationInfo
  parent: System.Runtime.Serialization
  isExternal: false
  name: SerializationInfo
  nameWithType: SerializationInfo
  fullName: System.Runtime.Serialization.SerializationInfo
- uid: System.Runtime.Serialization.StreamingContext
  parent: System.Runtime.Serialization
  isExternal: true
  name: StreamingContext
  nameWithType: StreamingContext
  fullName: System.Runtime.Serialization.StreamingContext
- uid: System.Workflow.Activities.Rules.RuleEvaluationException.#ctor(System.String,System.Exception)
  parent: System.Workflow.Activities.Rules.RuleEvaluationException
  isExternal: false
  name: RuleEvaluationException(String,Exception)
  nameWithType: RuleEvaluationException.RuleEvaluationException(String,Exception)
  fullName: System.Workflow.Activities.Rules.RuleEvaluationException.RuleEvaluationException(String,Exception)
- uid: System.Exception
  parent: System
  isExternal: true
  name: Exception
  nameWithType: Exception
  fullName: System.Exception
- uid: System.Workflow.Activities.Rules.RuleEvaluationException.#ctor*
  parent: System.Workflow.Activities.Rules.RuleEvaluationException
  isExternal: false
  name: RuleEvaluationException
  nameWithType: RuleEvaluationException.RuleEvaluationException
